internal/docs/design.md

```diff
@@ -184,6 +187,102 @@ needed). However, this would result in inconsistent names over time (e.g., some
 packages would expose `New` with an `Options`, while others would expose
 `NewWithOptions`).
 
+### URLs
+
+To enable the [Backing services factor][] of a Twelve-Factor Application, Go
+Cloud includes the ability to construct each of its API objects using
+identifying URLs. The concrete type's package should include APIs like the
+following:
+
+```go
+package foo
+
+// A type that implements WidgetURLOpener can open widgets based on a URL.
+// The opener must not modify the URL argument. OpenWidgetURL must be safe to
+// call from multiple goroutines.
```

Comment:
Is this required at this design doc level?

Reply:
The concurrency guarantee? Yes it is, since URL muxes may be shared at a global level.